src/zjs_callbacks.c

```diff
@@ -387,57 +418,112 @@ void print_callbacks(void)
 
 void zjs_call_callback(int32_t i)
```

Comment:
This function got ugly. We need to decide if we want to have both pre/post callbacks AND a ring buffer, or just the ring buffer.